src/ui/common/src/components/pages/workflows/index.tsx

```diff
@@ -149,14 +150,40 @@ const WorkflowsPage: React.FC<Props> = ({ user, Layout = DefaultLayout }) => {
         workflowId: workflowId,
       });
       let status = ExecutionStatus.Unknown;
-
+      let latestDagId;
+      let latestDagResultId;
       if (!dagResultsLoading && !dagResultsError && dagResults.length > 0) {
         const latestDagResult = getLatestDagResult(dagResults);
         if (latestDagResult) {
           status = latestDagResult.exec_state.status;
```

Comment:
I noticed that as we keep reassign this `status` variable, we will render different workflow status as these APIs are loading. This is a slight friction, and I think we can improve this by:
* Under `components/pages/workflow/hook` put together a helper like `useWorkflowStatus(workflow, dag, dagResult) -> status | undefined` that gather the status using these queries, and returns undefined if they are still loading. We can then render until the status is available